securesign/image-factory
# Add cosign to image factory

This pr is related to this issue [here](https://issues.redhat.com/browse/SECURESIGN-27), which involves adding a cosign image to the midstream image repository.

## Testing 
1. Check out Branch 
2. Run the GitHub actions either through GitHub or a tool like act.
3. Deploy the image to an open-shift cluster
4. Sign and verify a container image 

## Comments 
I was able to install the image on an open shift cluster and sign an image using both the public good instance as well as standing the stack up using our  helm charts. A point to note when signing you will need to curl the OIDC provider i.e key cloak like so: 

```
curl -X POST -H "Content-Type: application/x-www-form-urlencoded" -d "client_id=sigstore" -d "username=<username>" -d "password=<password>" -d "grant_type=password" -d "scope=openid" https://<Keycloak instance>/auth/realms/sigstore/protocol/openid-connect/token
```
This will require a change in the key cloak instance, you will need to ensure that Direct Access Grants Enabled, is enabled . This is done manually for now but can be configure using the helm chart. In the JSON response the field you will want is called id_token, An image can be then signed like so.

```
cosign sign -y --fulcio-url=$FULCIO_URL --rekor-url=$REKOR_URL --oidc-issuer=$KEYCLOAK_OIDC_ISSUER --identity-token=<id_token>  <image>
```

Cosign will also need to be authenticated with quay.io or any other image repo like so:

```
cosign login <repo_url> -u <username> -p <password>
```
